RELEASE.md

```diff
@@ -24,14 +24,12 @@ If you are interested in volunteering please create a pull request against the [
 
 ## Release shepherd responsibilities
 
-The release shepherd is responsible for the entire release series of a minor release, meaning all pre- and patch releases of a minor release. The process starts with the initial pre-release.
+The release shepherd is responsible for the entire release series of a minor release, meaning all pre- and patch releases of a minor release. The process formally starts with the initial pre-release, but some preparations should be done a few days in advance.
 
-* The first pre-release is scheduled according to the above schedule.
-* With the pre-release the release shepherd is responsible for running and monitoring a benchmark run of the pre-release for 3 days, after which, if successful, the pre-release is promoted to a stable release.
-* Once a pre-release has been released, the `master` branch of the repository is frozen for any feature work, only critical bug fix work concerning the minor release is merged.
-* Pre-releases are done from `master`, after pre-releases are promoted to the stable release a `release-major.minor` branch is created.
-
-_Experimental change of the above procedure for the v2.10 release: The `release-2.10` branch is created at the time the first pre-release is cut. All releases of the series including pre-releases are cut from the `release-2.10` branch. `master` will not be frozen for feature work._
+* We aim to keep the master branch in a working state at all times. In principle, it should be possible to cut a release from master at any time. In practice, things might not work out as nicely. A few days before the pre-release is scheduled, the shepherd should check the state of master. Following their best judgement, the shepherd should try to expedite bug fixes that are still in progress but should make it into the release. On the other hand, the shepherd may hold back merging last-minute invasive and risky changes that are better suited for the next minor release.
+* On the date listed in the table above, the release shepherd cuts the first pre-release (using the suffix `-rc.0`) and creates a new branch called  `release-<major>.<minor>` starting at the commit tagged for the pre-release. In general, a pre-release is considered a release candidate (that's what `rc` stands for) and should therefore not contain any known bugs that are planned to be fixed in the final release. If need be, the release has to be delayed.
```

Comment:
> should therefore not contain any known bugs that are planned to be fixed in the final release. If need be, the release has to be delayed.

This is not our standard practice, and risks us getting back into a place where releases are delayed months while one more bug needs to be fixed.

Reply:
I guess you mean that there can be known bugs which should not block a release. But isn't the "that are planned to be fixed in the final release" qualification sufficient to make sure it's only about a subset of known bugs?

Reply:
If we agree that a known bug is fine to have in the final release, it's OK to also to an RC with it. But if we have a bug we want to fix before the final release is out, we should in general not cut an RC that includes it. (Exceptions are possible, that's why the word "should" is used above.)

What I describe here is what I assume always has been our standard practice. Deviations were either deliberate exceptions or accidents.

What do all the others think?

Reply:
We've disobeyed this more often than obeyed, even 2.10 had a commit that wasn't in the rc - which is fine. What I'm worried about is that this indicates that it's more important to delay things to get them perfect, than to get a release out on a reasonable timeframe.

Reply:
We can soften the wording even more than "if need be". However, I don't want to convey the notion that you throw a buggy pile of mess at the users as a "release" candidate. That's not why we cut them. We cut them to discover unknown bugs before we do the final release .

Reply:
It's the final sentence with "delayed" that worries me more tbh.

Reply:
I got that. I'll just remove that sentence if that removes the controversy.